lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -4470,6 +4470,29 @@ TypeChecker::conformsToProtocol(Type T, ProtocolDecl *Proto, DeclContext *DC,
   return lookupResult;
 }
 
+bool
+TypeChecker::couldDynamicallyConformToProtocol(Type type, ProtocolDecl *Proto,
+                                               DeclContext *DC) {
+  // We may have types that can dynamically conform to a protocol, e.g. a non-final
+  // class which might have a subclass that conforms to the protocol, or standard
+  // library collection types such as Array, Set or Dictionary which have custom
+  // casting machinery implemented in situations like:
+  //
+  //  func encodable(_ value: Encodable) {
+  //    _ = value as! [String : Encodable]
+  //  }
```

Comment:
Sorry to nitpick, but IMO it would be clearer if the comment about the stdlib collections was written above the check for `isKnownStdlibCollectionType`, that way we can also explain that we're using `lookupConformance` to skip checking conditional requirements, as an intermediate collection cast can change whether or not the conditions are met.

Reply:
> Sorry to nitpick

No problem @hamishknight :) 

> that way we can also explain that we're using lookupConformance to skip checking conditional requirements, as an intermediate collection cast can change whether or not the conditions are met.

Great! I'll adjust 👍

Reply:
Fixed :)

Reply:
Thanks!